inc/item_operatingsystem.class.php

```diff
@@ -0,0 +1,379 @@
+      global $DB, $CFG_GLPI;
+
+      //default options
+      $params = ['rand' => mt_rand()];
+
+      $canedit = $item->canAddItem('OperatingSystem') && OperatingSystem::canView();
+
+      $columns = [
+         __('Name'),
+         __('Version'),
+         __('Architecture'),
+         __('Service pack')
+      ];
+
+      if (isset($_GET["order"]) && ($_GET["order"] == "ASC")) {
+         $order = "ASC";
+      } else {
+         $order = "DESC";
+      }
+
+      if ((isset($_GET["sort"]) && !empty($_GET["sort"]))
+         && isset($columns[$_GET["sort"]])) {
+         $sort = "`".$_GET["sort"]."`";
+      } else {
+         $sort = "`glpi_items_operatingsystems`.`id`";
+      }
+
+      if (empty($withtemplate)) {
+         $withtemplate = 0;
+      }
+
+      $query = "SELECT `glpi_items_operatingsystems`.`id` AS assocID,
+                       `glpi_operatingsystems`.`name`,
+                       `glpi_operatingsystemversions`.`name` AS version,
+                       `glpi_operatingsystemarchitectures`.`name` AS architecture,
+                       `glpi_operatingsystemservicepacks`.`name` AS servicepack
+                FROM `glpi_items_operatingsystems`
+                LEFT JOIN `glpi_operatingsystems`
+                           ON (`glpi_items_operatingsystems`.`operatingsystems_id`=`glpi_operatingsystems`.`id`)
+                LEFT JOIN `glpi_operatingsystemservicepacks`
+                           ON (`glpi_items_operatingsystems`.`operatingsystemservicepacks_id`=`glpi_operatingsystemservicepacks`.`id`)
+                LEFT JOIN `glpi_operatingsystemarchitectures`
+                        ON (`glpi_items_operatingsystems`.`operatingsystemarchitectures_id`=`glpi_operatingsystemarchitectures`.`id`)
+                LEFT JOIN `glpi_operatingsystemversions`
+                        ON (`glpi_items_operatingsystems`.`operatingsystemversions_id` = `glpi_operatingsystemversions`.`id`)
+                WHERE `glpi_items_operatingsystems`.`items_id` = '".$item->getID()."'
+                      AND `glpi_items_operatingsystems`.`itemtype` = '".$item->getType()."' ";
+
+      $query .= " ORDER BY $sort $order";
+
+      $result = $DB->query($query);
+      $number = $DB->numrows($result);
+      $i      = 0;
+
+      $os = array();
+      $used      = array();
+      if ($numrows = $DB->numrows($result)) {
+         while ($data = $DB->fetch_assoc($result)) {
+            $os[$data['assocID']] = $data;
+         }
+      }
+
+      $canedit = $item->canEdit($item->getID());
+
+      if ($canedit && $numrows >= 1
+          && !(!empty($withtemplate) && ($withtemplate == 2))) {
+         echo "<div class='center firstbloc'>".
+            "<a class='vsubmit' href='" . Toolbox::getItemTypeFormURL(self::getType()) . "?items_id=" . $item->getID() .
+            "&amp;itemtype=" . $item->getType() . "&amp;withtemplate=" . $withtemplate."'>";
+         echo __('Add an operating system');
```

Comment:
I've added the button Since the new database schema permit that, but this is not something I've planned to implement in the FI plugin. We can hide the button if you prefer, no problem.
Multiboot is not the norm anyways.